src/activitypub/fediverse-bridge.ts

```diff
@@ -36,6 +47,88 @@ export class FediverseBridge {
         );
     }
 
+    private async handlePostCreated(event: PostCreatedEvent) {
+        const post = event.getPost();
+        if (!post.author.isInternal) {
+            return;
+        }
+        const ctx = this.fedifyContextFactory.getFedifyContext();
+        let fedifyObject = null;
+
+        if (post.type === PostType.Note) {
+            fedifyObject = new FedifyNote({
+                id: post.apId || ctx.getObjectUri(FedifyNote, { id: uuidv4() }),
+                attribution: post.author.apId,
+                content: post.content,
+                summary: null,
+                published: Temporal.Now.instant(),
+                attachments: post.imageUrl
+                    ? [
+                          new Image({
+                              url: post.imageUrl,
+                          }),
+                      ]
+                    : undefined,
+                to: PUBLIC_COLLECTION,
+                cc: post.author.apFollowers,
+            });
+        } else if (post.type === PostType.Article) {
+            const preview = new FedifyNote({
+                id: ctx.getObjectUri(FedifyNote, { id: String(post.id) }),
+                content: post.excerpt,
+            });
+            fedifyObject = new Article({
+                id:
+                    post.apId ||
+                    ctx.getObjectUri(Article, { id: String(post.id) }),
+                attribution: post.author.apId,
+                name: post.title,
+                content: post.content,
+                image: post.imageUrl,
+                published: Temporal.Instant.from(
+                    post.publishedAt.toISOString(),
+                ),
+                preview,
+                url: post.url,
+                to: PUBLIC_COLLECTION,
+                cc: post.author.apFollowers,
+            });
+        }
+
+        const createActivity = new Create({
+            id: ctx.getObjectUri(Create, { id: uuidv4() }),
+            actor: post.author.apId,
+            object: fedifyObject,
+            to: PUBLIC_COLLECTION,
+            cc: post.author.apFollowers,
+        });
+
+        await ctx.data.globaldb.set(
+            [createActivity.id!.href],
+            await createActivity.toJsonLd(),
+        );
+
+        if (fedifyObject?.id!.href) {
```

Comment:
`fedifyObject` should always be defined - so we shouldn't need a `?` here, and we can get rid of the whole conditional!

If typescript tells us that it might not be defined, then we need to understand why and handle that case explicitly

Reply:
Fixed in https://github.com/TryGhost/ActivityPub/pull/551/commits/4a4a080f43440366183d9ed983a77132f8a21e9a